specfile/specfile.py

```diff
@@ -611,7 +611,12 @@ def add_changelog_entry(
                 if self.contains_autochangelog(section):
                     continue
                 if evr is None:
-                    evr = "%{?epoch:%{epoch}:}%{version}-%{release}"
+                    epoch_prefix = (
+                        f"{self.expanded_epoch}:" if self.expanded_epoch else ""
```

Comment:
Should this use the tags() context manager directly for better performance instead of two `expanded_*` properties that cause the tags context to be instantiated twice?

Reply:
Also, the values are expanded a second time on 623 which is probably not desired.

Reply:
I believe we're good with:
```python
                    with self.tags() as tags:
                        try:
                            evr = f"{tags.epoch.expanded_value}:"
                        except AttributeError:
                            evr = ""
                        evr += f"{tags.version.expanded_value}-"
                        evr += self.expand(
                            tags.release.value, extra_macros=[("dist", "")]
                        )
```

Marking as resolved.